File: AmeliorationQualite.cs
namespace exercice_trombone
{
    /// <summary>
    /// Effet d'amélioration de la qualité des trombones.
    /// </summary>
    internal class AmeliorationQualite : EffetAmelioration
    {
        /// <inheritdoc/>
        public override void AppliquerEffet(Usine usine)
        {
            usine.Qualite += valeur;
        }
    }
}

File: src/edu/cegepvicto/usine/AmeliorationQualite.java
package edu.cegepvicto.usine;

/**
 * Amélioration de la qualité de la production
 */
public class AmeliorationQualite extends EffetAmelioration {

    /**
     * @inheritDoc
     */
    @Override
    public void appliquerEffet(Usine usine) {
        usine.setQualite(usine.getQualite() + valeur);
    }
}
